tests/core/test_metrics_calculator.py

```diff
@@ -125,3 +127,31 @@ def test_column_partition_metric(
         )
         for idx, element in enumerate(results[desired_metric.id])
     )
+
+
+@mock.patch("great_expectations.execution_engine.execution_engine.ExecutionEngine")
+def test_get_metric(execution_engine: mock.MagicMock):
+    metrics_calculator = MetricsCalculator(execution_engine=execution_engine)
+    metric_name = "my_metric_name"
+    metric_domain_kwargs: dict = {}
+    with mock.patch(
+        "great_expectations.validator.metric_configuration.MetricConfiguration.metric_name",
+        new_callable=mock.PropertyMock,
+        return_value=metric_name,
+    ), mock.patch(
+        "great_expectations.validator.metric_configuration.MetricConfiguration.metric_domain_kwargs",
+        new_callable=mock.PropertyMock,
+        return_value=IDDict(metric_domain_kwargs),
```

Comment:
@Kilo59 Let us discuss to make sure we agree on the best alternative.  Thanks!

Reply:
I'm fine with what you have now, but if you want isolation from the real  `MetricConfiguation` you should be able to just use a namedtuple (or `IDDict`) in place of the real thing.
I still don't see the need for the `mock.patch` here.

Reply:
@Kilo59 The `MetricConfiguration` object is the one input to `get_metric()` -- the method under test.  So we need it, but its internals must be maximally mocked leaving only the aspects necessary for the test as un-mocked (to prevent test leakage).  I updated the docstring to emphasize this.  Thanks!